pixeltable/func/function.py

```diff
@@ -74,6 +66,16 @@ def call_return_type(self, kwargs: dict[str, Any]) -> ts.ColumnType:
                 kw_args[param.name] = bound_args.arguments[param.name]
         return self._call_return_type(**kw_args)
 
+    def dynamic_return_type(self, call_return_type: Callable[..., ts.ColumnType]) -> Callable[..., ts.ColumnType]:
```

Comment:
I actually think call_return_type is more specific (because it takes the actual call arguments into account)